### ISSUE TITLE
bug tracking - self destructs in 10 seconds

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [earthlng/testpages#2]
[bleep] [censored] [top secret] [redacted] [classified] [witchhunt] [russia]

**Comment by earthlng** [comment]
I don't have email notifications enabled :(

**Comment by Thorin-Oakenpants** [comment]
damnit - delete this when u see it

Open: https://bugzilla.mozilla.org/buglist.cgi?bug_status=__open__&email1=simon.mainey%40gmail.com&emailcc1=1&emailtype1=exact&list_id=13617441

Closed: https://bugzilla.mozilla.org/buglist.cgi?bug_status=__closed__&email1=simon.mainey%40gmail.com&emailcc1=1&emailtype1=exact&list_id=13617441

As I said earlier - its really hard to track "closed" items in the dashboard, and also by using my account as a flag, I can add just items we want to track. As I add closed items to stickies and the user.js I stop subscribing to them

**Comment by earthlng** [comment]
thanks, I bookmarked those 2 links - I knew that was your account :)
So that's how you keep track of all this stuff. I always wondered how you managed to do that.

**Comment by Thorin-Oakenpants** [comment]
so if you want anything added, just say so eg "Pants add ticket 123456"

**Comment by Thorin-Oakenpants** [comment]
eeek

![mixedweak](https://user-images.githubusercontent.com/16656956/34837276-26da6f36-f75f-11e7-96ad-fca85b4543dc.png)

**Comment by Thorin-Oakenpants** [comment]
broken encryption? scary icon in urlbar? Is this the `1262`'s? That's a nasty message.

**Comment by earthlng** [comment]
no it's because of the mixed passive content (images) that you recently flipped to true.

**Comment by Thorin-Oakenpants** [comment]
- I allowed `agrias.com.br` thru uBo so the test would work
- I only allowed css & images in uM
- user_pref("security.mixed_content.block_active_content", true);
- user_pref("security.mixed_content.block_display_content", true); << BLOCK DISPLAY CONTENT

So if Firefox has been told to block active and passive content on http sites - then why did uBo allow this

![logger](https://user-images.githubusercontent.com/16656956/34849318-de6e82d2-f786-11e7-9f00-9147477e36db.png)

Am I missing something here. How the F does uBo / uM allow this. Are they overriding the APP directive to block this stuff?

**Comment by earthlng** [comment]
> No, I didn't flip it, I changed it from active to inactive

https://github.com/ghacksuserjs/ghacks-user.js/commit/d82791a93347799898487adcabe827807abda753 ... you made it active and thus "flipped" it to true

> This is not good if extensions can bypass settings like this.

uh what? you said "uM did **NOT** have https-strict: * true" at the time of your testing, and afaik uBO doesn't have mixed-content blocking anyway, so what did you expect?

> As soon as I added https-strict: * true to uM, THEN it was blocked.

yes, working as expected. What you're seeing here is the priming stuff in action which is kinda the purpose of that whole testpage in the first place :) Once you block MC in uM, uM then interferes with the priming stuff. Even though both uBO and uM see and log the http image request to agrias.com.br, it actually never reaches the network because FF finds that agrias.com.br is in the HSTS preloadlist and "upgrades" the request to https. I'd imagine you could see that behaviour in action by checking with the built-in network manager in devtools.

**Comment by earthlng** [comment]
I think the test is broken atm anyway or they fundamentally changed something in the priming code. I can't even get the agrias image to load with **both** priming prefs set to true in my nightly.

update: 
https://dxr.mozilla.org/mozilla-central/source/security/manager/ssl/nsSTSPreloadList.errors#2058
agrias.com.br is now in this (new?) special list of (?) "domains with HSTS-preload errors" (?).

I'll update the page to use an image from a domain that's still in the preload list.

**Comment by earthlng** [comment]
ok now the 3rd image works again. From my testing in nightly, the broken encryption icon is caused by `security.mixed_content.use_hsts;true` and that's probably not the intended behaviour ie a bug.
At least the message shouldn't be that it "uses weak encryption" because when you load the image directly, the icon is green. And it doesn't need a message about broken mixed-content either because FF realizes that the domain is in the preloadlist and upgrades the request and it never actually requests the image via HTTP.

The 2nd image is broken too because they don't send the HSTS header anymore either. I'll try to find a suitable replacement.

**Comment by Thorin-Oakenpants** [comment]
nice PoC here for ya : https://bugzilla.mozilla.org/show_bug.cgi?id=1437349

**Comment by Thorin-Oakenpants** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1447226 -- took a whole WEEK .. suddenly P1

delete after reading

**Comment by Thorin-Oakenpants** [comment]
you still checking that email account I used that one time?

**Comment by earthlng** [comment]
I will if you drop me a note somewhere

**Comment by Thorin-Oakenpants** [comment]
forwarded u some emails

**Comment by earthlng** [comment]
new [Fullscreen API screen-res leak PoC](https://earthlng.github.io/testpages/fullscreen_leak.htm)

maybe let Arthur know, IDK. delete when done

**Comment by Thorin-Oakenpants** [comment]
I can't delete your stuff bro, read it, sending an email  - bcc'ing u in

delete when done

**Comment by Thorin-Oakenpants** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1450401

**Comment by Thorin-Oakenpants** [comment]
@earthlng Can you please your email. TIA :kiss:

**Comment by Thorin-Oakenpants** [comment]
Feel free to check your mail again. Do I need to keep saying this? Don;t you check it anyway, or am I only allowed a discarded throwaway email to GET AT YOU :trollface:

**Comment by earthlng** [comment]
it's not a throwaway, I just don't check it regularly

> Do I need to keep saying this?

if you want me to see it then yes ;)
I could check it once a week or so but I'll probably forget

**Comment by earthlng** [comment]
@Thorin-Oakenpants do you know what happened to your bugzilla lists? Did you delete your account or something?
The last few days I only get an error page

**Comment by Thorin-Oakenpants** [comment]
works for me - I'll send you an email, but they haven't changed

**Comment by earthlng** [comment]
thanks! I tried it again about an hour after your comment and it worked again for me too. IDK, must have been a server issue or something.

**Comment by Thorin-II** [comment]
they're tracking you bro .. amazon drones are in the air

**Comment by earthlng** [comment]
lol. have you seen the spaceX satellites in the night sky yet? I only saw them once when I went out for a smoke late at night.
I was equally shocked and amazed. kinda spooky :)

https://www.youtube.com/watch?v=l58nWIvyYs0

**Comment by Thorin-II** [comment]
not seen em, but fuck Elon Musk ruining my view

**Comment by earthlng** [comment]
don't worry, you'll see them when they have all **12000** planned satellites up there!!!

**Comment by Thorin-II** [comment]
fu....uuuuck ... I'll have to stick pebbles in my shoes to hide my gait, and mask up like trump, and wear tinfoil and reflectacles so they can't FP my biometrics .. maybe stop my heart beating or make it super irregular

**Comment by Thorin-II** [comment]
is that video time lapsed? Anyway, can;t see shit down here, we're too remote, practically on the edge of the flat disc

**Comment by earthlng** [comment]
> is that video time lapsed?

nope, that's pretty much how I saw it too. Only difference was that I only saw them in a limited space "window", like the ones at the front went out of view at a certain point and new ones came into view from the back. They were pretty low on the horizon and I guess the sun only reflected them into view in that limited space window

**Comment by Thorin-II** [comment]
so I basically reset my system fonts

here's RFP only in FF81 (note I have 9 styles in Arial, 6 in calibri)  .. 68 fonts detected
> Arial,  Arial Black, Arial Narrow, Calibri, Calibri Light, Calibri Light  Italic, Cambria, Cambria Math, Candara, Comic Sans MS, Consolas,  Constantia, Corbel, Courier, Courier New, Ebrima, Gabriola, Georgia,  Helvetica, Impact, Lucida Console, Lucida Sans Unicode, Malgun Gothic,  Marlett, Microsoft Himalaya, Microsoft JhengHei, Microsoft New Tai Lue,  Microsoft PhagsPa, Microsoft Sans Serif, Microsoft Tai Le, Microsoft  YaHei, Microsoft Yi Baiti, MingLiU_HKSCS-ExtB, MingLiU-ExtB, Mongolian  Baiti, MS Gothic, MS PGothic, ＭＳ Ｐゴシック, MS Sans Serif, MS Serif, MS UI  Gothic, ＭＳ ゴシック, MV Boli, NSimSun, Palatino Linotype, PMingLiU-ExtB,  Roman, Segoe Print, Segoe Script, Segoe UI, Segoe UI Light, Segoe UI  Semibold, Segoe UI Symbol, SimSun, SimSun-ExtB, Small Fonts, Sylfaen,  Symbol, Tahoma, Times, Times New Roman, Trebuchet MS, Twemoji Mozilla,  Verdana, Webdings, Wingdings, 宋体, 微软雅黑

what is the diff from yours: Arial Narrow and is there anything else?

Here's pantopticlick (so yeah, it picked up a few more like Consolas etc that you pointed out, now my system fonts are "normal")

> Arial, Arial Black, Arial Narrow, Calibri, Cambria, Cambria Math, Comic Sans MS, Consolas, Courier, Courier New, Georgia, Helvetica, Impact, Lucida Console, Lucida Sans Unicode, Microsoft Sans Serif, MS Gothic, MS PGothic, MS Sans Serif, MS Serif, Palatino Linotype, Segoe Print, Segoe Script, Segoe UI, Segoe UI Light, Segoe UI Semibold, Segoe UI Symbol, Tahoma, Times, Times New Roman, Trebuchet MS, Verdana, Wingdings

**Comment by earthlng** [comment]
> 68 fonts detected

I get 65

> what is the diff from yours

I don't have Arial Narrow, Calibri Light, Calibri Light Italic. The other 65 are the same as yours

**Comment by Thorin-II** [comment]
hah, I just guessed as much in the other thread

**Comment by Thorin-II** [comment]
It looks pretty solid to me if there's only a few extra styles to create entropy. 220 million FF users with RFP on, one day .. ... maybe ... Of course it would be better if they move away from families and actually use individual styles

**Comment by earthlng** [comment]
> Of course it would be better if they move away from families and actually use individual styles

yeah totally. It looks like if a user or another software installs additional styles of a font in the allowlist then that gets allowed too. That should not be happening. Fe in your case, Arial Narrow is not in the list and should not be allowed.

**Comment by Thorin-II** [comment]
I don't know how all that plays out in Linux or Mac, but I think we could open a ticket with some evidence: i.e issue at user.js and ask windows users to post results - and then you and I can keep a list of diffs - e.g. me being a control set, or something?

edit: we would post a hash and detail the results .. and ask users only to post both hash and the list if it's a new hash - capsice?

**Comment by Thorin-II** [comment]
A ticket at bugzilla for jfkthame to follow up on his ToDo on line 7

**Comment by earthlng** [comment]
sounds good 👍

**Comment by Thorin-II** [comment]
hah, so it's easy to detect if  you're gecko based (< 1ms), its easy to detect if you're running RFP (1ms), and now (assuming a whitelist isn't used, and obviously you allow for doc fonts blocked) it's probably easy to infer Win7 vs Win10 based on some fonts: although you could have done that anyway, I guess.

They're not going to bother with win7 - I suspect the number of users will drop and they'll end support maybe in the next ESR (probably the next one after: i.e in two years)

I can see the list being tightened up in several ways
 - drop some most-likely not needed crap that causes entropy (i.e move into kLangPacks or create another section, or drop them (comment them out)), like `HoloLens MDL2 Assets` (there's another MDL2 there but they both have it: seems part of a family)
 - reduce the number (this is not to make win7 and win10 more alike, just to eliminate potential entropy within win10): e.g. I'm sure RFP users can get along without some of those fonts (win7 doesn't have Bahnschrift or all those Sitka / Yu Gothic sets for example and we get along fine)
 - family styles <-- this

Only some decent data will tell the story. Moz are collecting telemetry on this. Wish I could get a data set
